strawberryfields/apps/qchem/dynamics.py

```diff
@@ -250,3 +254,62 @@ def prob(samples: list, excited_state: list) -> float:
         raise ValueError("The excited state must not contain negative values")
 
     return samples.count(excited_state) / len(samples)
+
+
+def marginal(mu: np.ndarray, V: np.ndarray, n: int, hbar: float = 2.0) -> np.ndarray:
+    r"""Generate marginal distributions from the displacement vector and covariance matrix of a
+    state.
+
+    **Example usage:**
+
+    >>> mu = np.array([0.00000000, 2.82842712, 0.00000000,
+    >>>                0.00000000, 0.00000000, 0.00000000])
+    >>> V = np.array([[1.0, 0.0, 0.0, 0.0, 0.0, 0.0],
+    >>>               [0.0, 1.0, 0.0, 0.0, 0.0, 0.0],
+    >>>               [0.0, 0.0, 1.0, 0.0, 0.0, 0.0],
+    >>>               [0.0, 0.0, 0.0, 1.0, 0.0, 0.0],
+    >>>               [0.0, 0.0, 0.0, 0.0, 1.0, 0.0],
+    >>>               [0.0, 0.0, 0.0, 0.0, 0.0, 1.0]])
+    >>> n = 10
+    >>> marginal(mu, V, n)
+    array([[1.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+            0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+            0.00000000e+00, 0.00000000e+00],
+           [1.35335284e-01, 2.70670567e-01, 2.70670566e-01, 1.80447044e-01,
+            9.02235216e-02, 3.60894085e-02, 1.20298028e-02, 3.43708650e-03,
+            8.59271622e-04, 1.90949249e-04],
+           [1.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+            0.00000000e+00, 0.00000000e+00, 0.00000000e+00, 0.00000000e+00,
+            0.00000000e+00, 0.00000000e+00]])
+
+    Args:
+        mu (array): displacement vector
+        V (array): covariance matrix
+        n (int): number of vibrational states
+        hbar (float): the value of :math:`\hbar` in the commutation relation :math:`[\x,\p]=i\hbar`.
+
+    Returns:
+        array[list[float]]: marginal distributions
+    """
+    if not V.shape[0] == V.shape[1]:
+        raise ValueError("The covariance matrix must be a square matrix")
+
+    if not len(mu) == len(V):
+        raise ValueError(
+            "The number of modes in the displacement vector and the covariance matrix"
```

Comment:
```suggestion
            "The dimension of the displacement vector and the covariance matrix"
```

Reply:
Since the dimension of these is 2* nr_modes, so it's confusing to talk about the number of modes in them